openedx/features/content_type_gating/block_transformers.py

```diff
@@ -62,7 +62,9 @@ def _get_block_group_access(block_structure, block_key):
             block_key, UserPartitionTransformer, 'merged_group_access', None
         )
         if merged_access:
-            current_access = merged_access.get_allowed_groups()
+            # merged_access holds a dictionary of sets, but group_access is a dictionary of lists, so we convert here
+            # (sets seem like a better format for this, but existing code already expects lists)
+            current_access = {p: list(g) for (p, g) in merged_access.get_allowed_groups().items()}
```

Comment:
why was this change necessary?

Reply:
If you look lower in that file (and elsewhere in platform), you'll see a lot of code that compares `group_access` items directly to some list of permissions. If group_access holds sets instead of lists, those comparisons will always fail.

As the comment above this line says, for some reason, the `merged_access` field holds sets, not lists. So we convert.